rdflib/graph.py

```diff
@@ -1070,13 +1072,24 @@ def parse(
         )
         if format is None:
             format = source.content_type
+        assumed_xml = False
         if format is None:
-            # raise Exception("Could not determine format for %r. You can" + \
-            # "expicitly specify one with the format argument." % source)
-            format = "application/rdf+xml"
+            try:
+                from rdflib.util import guess_format  # local import avoids circular dependency
+                format = guess_format(source.file.name)
+            finally:
```

Comment:
It is not clear why this is wrapped in a try/finally block. Any error that occurs inside the try block will be fatal and parsing will not be able to continue.

Reply:
Thanks for pointing this out. I somehow brain-farted on the semantics of try/except/finally. I pushed what should hopefully fix this -- I catch specific exceptions that are to be expected.